File: java-automation/src/main/java/com/automation/pages/InventoryPage.java
package com.automation.pages;

import com.microsoft.playwright.Page;
import com.microsoft.playwright.Locator;
import com.automation.data.TestData;

import static org.junit.jupiter.api.Assertions.*;

public class InventoryPage {
    private final Page page;

    public InventoryPage(Page page) {
        this.page = page;
    }

    private Locator getProductNames() {
        return page.locator(".inventory_item_name");
    }

    private Locator getProductPrices() {
        return page.locator(".inventory_item_price");
    }

    private Locator getCartBadge() {
        return page.locator(".shopping_cart_badge");
    }

    private Locator getCartLink() {
        return page.locator(".shopping_cart_link");
    }

    private Locator getSortDropdown() {
        return page.locator("[data-test='product-sort-container']");
    }

    public void goTo() {
        page.navigate(TestData.Urls.INVENTORY);
    }

    public void addToCartByDataTest(String dataTestId) {
        page.locator("[data-test='" + dataTestId + "']").click();
    }

    public void goToCart() {
        getCartLink().click();
        assertEquals(TestData.Urls.CART, page.url(), "User should be redirected to cart page");
    }

    public void assertProductCount(int expected) {
        assertEquals(expected, getProductNames().count(), 
            "Expected " + expected + " products but found " + getProductNames().count());
    }

    public void assertCartBadgeCount(int expected) {
        assertEquals(String.valueOf(expected), getCartBadge().textContent(), 
            "Cart badge should show " + expected + " items");
    }

    public void selectSort(String option) {
        getSortDropdown().selectOption(option);
    }
}

File: python-automation/src/automation/pages/inventory_page.py
from playwright.sync_api import Page
from src.automation.data.test_data import Urls


class InventoryPage:
    def __init__(self, page: Page):
        self.page = page

    @property
    def product_names(self):
        return self.page.locator(".inventory_item_name")

    @property
    def product_prices(self):
        return self.page.locator(".inventory_item_price")

    @property
    def cart_badge(self):
        return self.page.locator(".shopping_cart_badge")

    @property
    def cart_link(self):
        return self.page.locator(".shopping_cart_link")

    @property
    def sort_dropdown(self):
        return self.page.locator("[data-test='product-sort-container']")

    def go_to(self):
        self.page.goto(Urls.INVENTORY)

    def add_to_cart_by_data_test(self, data_test_id: str):
        self.page.locator(f"[data-test='{data_test_id}']").click()

    def go_to_cart(self):
        self.cart_link.click()
        assert self.page.url == Urls.CART, \
            f"User should be redirected to {Urls.CART}, but got {self.page.url}"

    def assert_product_count(self, expected: int):
        count = self.product_names.count()
        assert count == expected, \
            f"Expected {expected} products but found {count}"

    def assert_cart_badge_count(self, expected: int):
        badge_text = self.cart_badge.text_content()
        assert badge_text == str(expected), \
            f"Cart badge should show {expected} items but shows {badge_text}"

    def select_sort(self, option: str):
        self.sort_dropdown.select_option(option)
